src/my-bot.ts

```diff
@@ -1,38 +1,423 @@
+import * as firebase from 'firebase';
+import { Position } from './interfaces/position';
+import { ShipPlace } from './interfaces/shipPlace';
+import { Shot } from './interfaces/shot';
+
 export class MyBot {
-    public getShipPositions() {
-        return [
-            { StartingSquare: { Row: "A", Column: 1 }, EndingSquare : { Row: "A", Column: 5 } },
-            { StartingSquare: { Row: "C", Column: 1 }, EndingSquare : { Row: "C", Column: 4 } },
-            { StartingSquare: { Row: "E", Column: 1 }, EndingSquare : { Row: "E", Column: 3 } },
-            { StartingSquare: { Row: "G", Column: 1 }, EndingSquare : { Row: "G", Column: 3 } },
-            { StartingSquare: { Row: "I", Column: 1 }, EndingSquare : { Row: "I", Column: 2 } },
-        ]
-    }
 
-    public selectTarget(gamestate) {
-        var previousShot = gamestate.MyShots && gamestate.MyShots[gamestate.MyShots.length-1];
-        if(previousShot) {
-            return this.getNextTarget(previousShot.Position);
+    private config = {
+        apiKey: "AIzaSyD6ACQdu7gK-BgtJs-3Hu1Lkczk8fp0Abo",
+        authDomain: "brokenbot-battleships.firebaseapp.com",
+        databaseURL: "https://brokenbot-battleships.firebaseio.com",
+        storageBucket: "brokenbot-battleships.appspot.com"
+    };
+    private matchId: number;
+
+    constructor() {
+        firebase.initializeApp(this.config);
+        this.authenticate();        
+    }
+    
+    public getShipPositions(): ShipPlace[] {
+        let exists: boolean = false;
+        let counter: number = 0;        
+        do {
+            this.matchId = Math.floor(Math.random() * 10000) + 1;
+            firebase.database().ref('matches/' + this.matchId.toString()).once('value').then((snapshot) => {
+                if (snapshot.val()) {
+                    exists = true;
+                } else {
+                    exists = false;
+                }
+            });
+            counter++;
+            if (counter > 10000) {
+                throw new Error("Infinite loop when constructing bot");
+            }
+        } while (exists)
+        firebase.database().ref('matches/' + this.matchId.toString()).set({
+            started: true,
+            hitmode: false,
+            sizes: {
+                patrol: true,
+                submarine: true,
+                destroyer: true,
+                battleship: true,
+                carrier: true
+            },
+            hitmap: []
+        });

```

Comment:
To be honest, I would pull out all of these firebase methods onto their own class - that way another dev knows that that's the interface with firebase